### PR TITLE
Rewriting BinomialTestBayesian with JaspResults

### PR DESCRIPTION
**Pull request by gibbona1**
binomialtestbayesian.R
  Updated Bayesian Binomial Test module to JASP results
  Added the descriptives plots option as in the regular Binomial Test module

BinomialTestBayesian.qml
  now using jaspResults
  Added descriptives plot option

PlotRobustnessSequential.R
  Fixed subscripting for Bayes factor, will now show correctBayes factor labeling in the top left
  Fixed subscripting for hypothesis labels, will now show correct hypothesis label for evidence in the top right, as well as evidence arrows
  Fixed an error that occurred with "very strong" label

### CODE REVIEW

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 1]
```diff
@@ -15,497 +15,239 @@
...
+  return(results)
+}
+
+# Main Table ----
+.bayesianBinomTableMain <- function(jaspResults, dataset, options, bayesianBinomResults, errors){
+  if (!is.null(jaspResults[["bayesianBinomialTable"]])) return()
+  variables <- unlist(options$variables)
+  theta0<- .5
+  
+  bayesianBinomialTable                   <- createJaspTable("Bayesian Binomial Test")
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]: Please fix the spacing here and elsewhere. In general, there should be a single space before and after binary operators (`+`, `-`, `==`, etc.) and assignment operators (`<-`). Exceptions are for aligning multiple lines.

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 2]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
+    #	BF10 <- NA
+    #	
+    #} else {
+    
+    BF10 <- try(.bayesBinomialTest.oneSided(counts, n, theta0, a, b, hypothesis), silent = TRUE)
+    
+    #}
+  }
+  
+  if (class(BF10) == "try-error")
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]:
```suggestion
  if (isTryError(BF10))
```

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 3]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
+.dfLinesPP <- function(dataset, a = 1, b = 1, hyp = "two-sided", theta0=.5, counts, n){
+  if (a == 1 && b == 1) {
+    
+    theta <- seq(0, 1, length.out = 1000)
+    
+  } else {
+    
+    theta <- seq(0.001, 0.999, length.out = 1000)
+  }
+  x <- theta
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]: this reassignment is confusing and I think redundant? You could just work with theta directly.

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 4]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
+  if      (hyp == "two.sided"){
+    linesGroup <- c(dbeta(x, a + counts, b + n - counts), dbeta(x, a, b))
+  }
+  else if (hyp== "greater")   {
+    linesGroup <- c(ifelse (x >= theta0,
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]: the `ifelse` statements are a little bit confusing (and inefficient). How about something like:

```r
size <- length(x)
linesPrior <- linesPosterior <- numeric(size) # initializes everything to 0
idx <- x >= theta0 # the nonzero components
# fill in the posterior
linesPosterior[idx] <- dbeta(x[idx], a + counts, b + n - counts) / pbeta(theta0, a + counts, b + n - counts, lower.tail = FALSE)
# fill in the prior
linesPrior[idx] <- dbeta(x[idx], a, b) / pbeta(theta0, a, b, lower.tail = FALSE)
linesGroup <- c(linesPosterior, linesPrior)
```

same goes for the `hyp == "less"` case. Then it becomes very clear that the only difference between `hyp == "greater" and `hyp == "less"` is whether `idx <- x >= theta0` or `idx <- x <= theta0`, and you could simplify the statement.

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 5]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
-			} else if (hypothesis == "less") {
-				text(xxT, yyT, paste("Evidence for H-:\n", lab), cex = 1.4, pos = 2, offset = -.2)
-			}
-		}
-		
-		if (BF10e < 1)
-			text(xxT, yyT, paste("Evidence for H0:\n", lab), cex = 1.4, pos = 2, offset = -.2)
-	}
+.dfLinesSR <- function(d, var, level, split, a = 1, b = 1, hyp = "two-sided", theta0=.5){
+  x <- ifelse (d == split, 1, 0)
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]:
```suggestion
  x <- 1 * (d == split)
```

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 6]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
+  BF10 <- vector("numeric", length(x))
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]:
```suggestion
  BF10 <- numeric(length(x))
```

File: JASP-Engine/JASP/R/binomialtestbayesian.R [hunk 7]
```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
...
+  for (i in seq_along(x)) {
+    
+    counts <- sum(x[1:i] == 1)
+    n <- length(x[1:i])
```
Comment (vandenman) [inline JASP-Engine/JASP/R/binomialtestbayesian.R]: maybe
```suggestion
    n <- i
```
?

**Comment by vandenman** [comment]
@TimKDJ Travis is unhappy about the figures. However, on @gibbona1 's branch they pass locally. If I checkout the branch, they do fail. But if I remake the .svg (so they pass locally for me) then they still fail on Travis ([branch](https://github.com/vandenman/jasp-desktop/tree/branc-mycommits), [travis log](https://github.com/vandenman/jasp-desktop/tree/branc-mycommits)). Any suggestions on how to proceed?

**Comment by TimKDJ** [comment]
1. Make sure both JASPgraphs & jaspResults are up-to-date with the current dev branch code
2. Don't use [[]] indexing if there are multiple graphs; the order in which they are stored in the state is random

**Comment by vandenman** [comment]
@TimKDJ thanks for the quick reply! @gibbona1 can you add non-numeric indexing?

**Comment by vandenman** [comment]
@TimKDJ indexing the plots with "1.png" doesn't work because these names always change in consecutive runs.. maybe you have time tomorrow to look at this?

**Comment by TimKDJ** [comment]
You get the name from the results ("data" property) and use that to get the correct plot from the state.

**Comment by vandenman** [comment]
@gibbona1 that could work, can you check? @TimKDJ almost no test looks up figures in this way (looking at anova, ancova, descriptives, linear regression, correlation). Reason for concern?

**Comment by TimKDJ** [comment]
It's only a problem for jaspResults analyses that obtain several plots after run().
Normal analyses' plot order is deterministic.

**Comment by gibbona1** [comment]
@TimKDJ @vandenman  if I go to
results[["results"]][["bayesianBinomPlots"]][["collection"]][["bayesianBinomPlots_deeperBayesianBinomPlotscontBinom2"]][["collection"]][["bayesianBinomPlots_deeperBayesianBinomPlotscontBinom2_contBinom2sequential"]][["data"]]
it gives me the "3.png" and "4.png" as before,
However if I rerun the line 
results <- jasptools::run("BinomialTestBayesian", "test.csv", options)
both the figure names in state and the data names in results (above) increment to something like "15.png" and "16.png"

I tried using the "name" for each, like
"bayesianBinomPlots_deeperBayesianBinomPlotscontBinom2_contBinom2sequential" 
but no luck.

**Comment by TimKDJ** [comment]
That's because the names are dynamic.. 
See my response here: https://github.com/jasp-stats/INTERNAL-jasp/issues/328

**Comment by TimKDJ** [comment]
You can keep your changes as is, but with #3376 the old style indexing works again

**Comment by TimKDJ** [comment]
Can you rebase?

**Comment by vandenman** [comment]
another rebase should be enough to please travis (see https://github.com/vandenman/jasp-desktop/tree/testBranch).
So once more 
```
git fetch upstream
git rebase upstream/master
git push -f
```

**Comment by vandenman** [comment]
uh oh, something bad happened and you rewrote the last five years of JASP's history! 

If you prefer I can also make this PR from https://github.com/vandenman/jasp-desktop/commits/testBranch, which should have the same changes as you had and is up to date with jasp-desktop. That way you don't have to worry about the git mess

**Comment by gibbona1** [comment]
@vandenman I used the code from your above comment and yeah It rewrote everything... 
if you could make the pull request from your branch that would be great.
Is there anything I need to undo here?

**Comment by TimKDJ** [comment]
Yeah rebasing on the master is not a good idea..

**Comment by vandenman** [comment]
uh oh that should have been a rebase on development not master, sorry my bad!

**Comment by TimKDJ** [comment]
For future reference: you can git reset to undo an action (you can find the action you want to undo in the reflog)

**Comment by vandenman** [comment]
can you still undo a force push? because I thought that rewrites history (and really means there's no way back..)

**Comment by TimKDJ** [comment]
Yeah it's still possible to do locally :)

**Pull request merged**